app/src/main/java/com/riadh/movies/ui/BaseActivity.java

```diff
@@ -0,0 +1,198 @@
+package com.riadh.movies.ui;
+
+
+import android.support.annotation.NonNull;
+import android.support.v7.app.AppCompatActivity;
+import android.text.TextUtils;
+import android.view.MenuItem;
+
+import com.afollestad.materialdialogs.DialogAction;
+import com.afollestad.materialdialogs.MaterialDialog;
+import com.google.gson.Gson;
+import com.riadh.movies.R;
+import com.riadh.movies.app.Constants;
+import com.riadh.movies.models.ErrorResponse;
+import com.riadh.movies.service.NetworkRetryRequest;
+import com.riadh.movies.utils.DialogUtil;
+
+import java.io.IOException;
+
+import okhttp3.ResponseBody;
+import retrofit2.Response;
+
+public class BaseActivity extends AppCompatActivity {
+
+
+    private MaterialDialog progressDialog;
+
+
+    protected void configActionBar(boolean displayHomeButton) {
+        if (getSupportActionBar() != null) {
+            getSupportActionBar().setDisplayHomeAsUpEnabled(displayHomeButton);
+            getSupportActionBar().setDisplayShowHomeEnabled(displayHomeButton);
+            getSupportActionBar().setDisplayShowTitleEnabled(displayHomeButton);
+        }
+    }
+
+    protected void configActionBar(boolean displayHomeButton, String title) {
+        if (getSupportActionBar() != null) {
+            getSupportActionBar().setDisplayHomeAsUpEnabled(displayHomeButton);
+            getSupportActionBar().setDisplayShowHomeEnabled(displayHomeButton);
+            getSupportActionBar().setDisplayShowTitleEnabled(displayHomeButton);
+            getSupportActionBar().setTitle(title);
+        }
+    }
+
+    public void configActionBarWithTitle(int title) {
+        if (getSupportActionBar() != null) {
+            getSupportActionBar().setDisplayHomeAsUpEnabled(false);
+            getSupportActionBar().setDisplayShowHomeEnabled(false);
+            getSupportActionBar().setDisplayShowTitleEnabled(true);
+            getSupportActionBar().setTitle(title);
+        }
+    }
+
+    public void showProgressDialog() {
+        if (progressDialog == null) {
+            progressDialog = DialogUtil.getDefaultProgressBar(this).build();
+        }
+        try {
+            progressDialog.show();
+        } catch (Exception ignored) {
+
+        }
+    }
+
+
+    public void hideProgressDialog() {
+        if (progressDialog == null) {
+            progressDialog = DialogUtil.getDefaultProgressBar(this).build();
+        }
+        try {
+            progressDialog.dismiss();
+        } catch (Exception ignored) {
+
+        }
+    }
+
+
+    public void handleUnexpectedError(Throwable t) {
+        String message = t.getMessage();
+        if (message == null)
```

Comment:
In other parts of the code you have used brackets for single line if, why didnt  you do the same here?

Reply:
I always use brackets in my code , in this line i just forgot it.